### PR TITLE
chore(client): document joining a federation

### LINKED ISSUE #4397: Add network to fedimint invite code
It is non-obvious how to check for the network when joining a federation. The current way requires creating a wallet client and asking what the network is after you've connected to the federation. Would be nice if you check up font without needing to connect.

I imagine this was never done because the network is really at the module level rather than the federation level but would be nice to be able to have some indication.

### PR DESCRIPTION
**Pull request by elsirion**
Hopefully fixes #4397 (mostly a documentation issue imo). @benthecarman would this have helped?

Screenshot 2024-02-23 at 18-16-44 ClientBuilder in fedimint_client - Rust

### CODE REVIEW

**Comment by benthecarman** [comment]
That's look good one! One of thing we ran into was downloading the config every start up, I think these docs would make people think they need to do a similar thing.

**Comment by elsirion** [comment]
I think  #4326 will land before this, then I'll have to rebase/fix the docs anyway and can add something about `open`.

**Comment by elsirion** [comment]
I added the comment about `open`.

File: fedimint-client/src/lib.rs [hunk 1]
```diff
@@ -1792,6 +1792,11 @@ impl ClientBuilder {
 
     /// Join a new Federation
     ///
+    /// When a user wants to connet to a new federation this function fetches
```
Comment (bradleystachurski) [inline fedimint-client/src/lib.rs:1795]: [nit] s/connet/connect

**Review by bradleystachurski — APPROVED** [review]
Looks good!

File: fedimint-client/src/lib.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-client/src/lib.rs]: Thx, fixed :)

**Review by bradleystachurski — APPROVED** [review]

**Review by dpc — APPROVED** [review]

**Pull request merged**